### PR TITLE
Create impervious_surface.py

### PR DESCRIPTION
**Pull request by tedw0ng**
Adding layer for impervious surface from GEE Tsinghua dataset.

### CODE REVIEW

File: city_metrix/layers/impervious_surface.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class ImperviousSurface(Layer):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def get_data(self, bbox):
+        # load impervious_surface
+        dataset = ee.Image("Tsinghua/FROM-GLC/GAIA/v10").gt(0)  # change_year_index is zero if permeable as of 2018
+        imperv_surf = ee.ImageCollection(dataset
```
Comment (weiqi-tori) [inline city_metrix/layers/impervious_surface.py:16]: `dataset` needs to be forced to ImageCollection to apply the `filterBounds()` and other functions.

`AttributeError: 'Image' object has no attribute 'filterBounds'`

**Review by weiqi-tori — CHANGES_REQUESTED** [review]
Add a test function to check if the new layer works well

File: city_metrix/layers/impervious_surface.py (hunk 1 above)
Comment (tedw0ng) [reply city_metrix/layers/impervious_surface.py:16]: Forced dataset to be imagecollection, and added test for impervious surface. (Not 100% sure I did that right.)

File: tests/test_layers.py [hunk 2]
```diff
@@ -77,6 +78,9 @@ def test_high_land_surface_temperature():
     data = HighLandSurfaceTemperature().get_data(BBOX_BRAZIL_LAURO_DE_FREITAS_1)
     assert data.any()
 
+def test_impervious_surface():
+    data = ImperviousSurface().get_data(BBOX_BRAZIL_LAURO_DE_FREITAS_1)
+	assert data.any()
```
Comment (weiqi-tori) [inline tests/test_layers.py:83]: Indent error cause test failed

**Comment by weiqi-tori** [comment]
The structure of `test_layers.py` has been updated recently, the test function needs to be updated accordingly to pass the test. You could rebase on the main branch first, then make the adjustment.
`FAILED tests/test_layers.py::test_impervious_surface - NameError: name 'BBOX_BRAZIL_LAURO_DE_FREITAS_1' is not defined`

**Review by weiqi-tori — APPROVED** [review]

**Pull request merged**